### DISCUSSION TITLE
moving online stuff to online plugin

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#4970] [Ideas]
My question is: if you don't have a community site (with user interaction), only a business one, you don't need to know who is online.  

Is there any reason why this can't be solved with the online plugin (except code rewriting) and online stuff moved out of core?

Because e107 can already be running with "no-online" mode.  

I wouldn't ask if I already didn't try this and it was too easy for my liking.  Maybe I am missing something. 

This is what I did:
- not using e_online handler directly
- added this handler only if the online plugin is installed this way in e107_class.php

```
		if(file_exists(e_ROOT . $e107_paths['PLUGINS_DIRECTORY'] . "online/handlers/online_class.php" )) {
			self::$_known_handlers['e_online'] =  '{e_PLUGIN}online/handlers/online_class.php';
		} 
```

and in class2.php
```
if(!isset($_E107['no_online']))
{
	if(e107::isInstalled('online')) {
		e107::getOnline()->goOnline($pref['track_online'], $pref['antiflood1']);
	}
	else {
		$_E107['no_online'] = true;
	} 
 
}
```
and move the dashboard from the user plugin to the online plugin.

Everything works. 

Did I miss something or is e107 so brilliant?  

Thanks

**Comment by Deltik** [comment]
Yes, it makes sense to move the online users handler to the plugin that's already there.  The e107 core and the e107 core plugins are not cleanly separated, but they should be.  It's not always easy due to legacy usages, but we can add deprecations in `e107::getOnline()` and other "core" handlers that should be in plugins instead.

You might remember that I previously had trouble with this in https://github.com/e107inc/e107/issues/4199.  With e_online, it's significantly easier to separate; someone just has to update the code to do it.  It's fairly low in priority, though, because it doesn't change much in behavior or fix any major bugs.
